src/block.cxx

```diff
@@ -0,0 +1,93 @@
+#include "block.h"
+
+#include "math/nemlmath.h"
+
+namespace neml {
+
+int block_evaluate(
+    std::shared_ptr<NEMLModel> model,
+    size_t nblock,
+    const double * const e_np1, const double * const e_n,
+    const double * const T_np1, const double * const T_n,
+    double t_np1, double t_n,
+    double * const s_np1, const double * const s_n,
+    double * const h_np1, const double * const h_n,
+    double * const A_np1,
+    double * const u_np1, const double * const u_n,
+    double * const p_np1, const double *  p_n)
+{
+  // I am (apparently) willing to trade memory for time
```

Comment:
this trade-off is not surprising... no action-needed

Reply:
Yep, just a reminder to me in the future that I'm doubling the memory requirement (well nearly) by adding the extra arrays.